src/systemctl/systemctl-list-unit-files.c

```diff
@@ -157,8 +157,8 @@ int verb_list_unit_files(int argc, char *argv[], void *userdata) {
 
                 n_units = hashmap_size(h);
 
-                units = new(UnitFileList, n_units ?: 1); /* avoid malloc(0) */
-                if (!units)
+                units = new(UnitFileList, n_units);
+                if (!units && n_units > 0)
```

Comment:
Actually we can ignore OOM error when `n_units == 0`, but even in that case, we usually call `log_oom()` and fail earlier. Please drop the added condition.

Reply:
Done.